Lib/test/test_sqlite3/test_dbapi.py

```diff
@@ -650,6 +665,16 @@ def __getitem__(slf, x):
         with self.assertRaises(ZeroDivisionError):
             self.cu.execute("select name from test where name=?", L())
 
+    def test_execute_too_many_params(self):
+        category = sqlite.SQLITE_LIMIT_VARIABLE_NUMBER
+        msg = "too many SQL variables"
+        with cx_limit(self.cx, category=category, limit=2):
+            self.cu.execute("select * from test where name=? and income=?",
+                            ("a", 1))
+            with self.assertRaisesRegex(sqlite.OperationalError, msg):
+                self.cu.execute("insert into test values(?, ?, ?)",
+                                ("a", 2, "b"))
```

Comment:
```suggestion
        with cx_limit(self.cx, category=category, limit=1):
            self.cu.execute("select * from test where name=?", ("a",))
            with self.assertRaisesRegex(sqlite.OperationalError, msg):
                self.cu.execute("select * from test where name=? and income=?",
                                ("a", 1))
```

Reply:
I made these a little less noisy in 6197b3a8ab5e9834ca0efb86d0b1beddc4d1ab7f